scripts/o3de/o3de/repo_properties.py

```diff
@@ -449,6 +522,8 @@ def add_parser_args(parser):
                                help='Prints the anticipated changes to your repo.json file object fields without actually writing to repo.json file.')
     general_group.add_argument('--force', '-f', action='store_true', default=False,
                                    help='Overwrite the release-archive zip file if there is already an existing zip with the same name.')
+    general_group.add_argument('--copy-to-repo-subfolder', '-cr', action='store_true', default=False,
```

Comment:
I think this argument should let the user provide a subfolder to copy the object into, or use a default if they provide no subfolder.  Otherwise, it would be impossible to use this command to copy an object into the repo root, if it is the only object in the repo.  For example, what if the user wanted to copy a gem at `c:\gem` into `c:\repo` such that the root of the gem is `c:\repo` and the `gem.json` file is at `c:\repo\gem.json`?

One way it to let the user provide a path for the copy argument.
e.g. if the repo is at `c:/repo`
`--cr ""` would copy the object into `c:/repo/`
`--cr Custom` would copy it into the `c:/repo/Custom` folder
`--cr` would default to copy the object into `c:/repo/<ObjectType>/<ObjectName>` folder

Thoughts?

Reply:
Sounds good, I currently only have it default copy to the associated object folder

Reply:
+1
This _could_ be a separate PR as adding this wouldn't break the existing API

Reply:
Heads up:
argparse interprets both --cr and --cr "" to be the same value.
In other words if "" is used, it'll fallback to the defined "const" value.